cpp/include/cudf/detail/utilities/cuda.cuh

```diff
@@ -25,8 +24,7 @@
 #include <rmm/cuda_stream_view.hpp>
 
 #include <cub/cub.cuh>
-
-#include <type_traits>
+#include <cuda/std/type_traits>
```

Comment:
To actually use this header I think we need to replace `std::is_arithmetic_v` with `cuda::std::is_arithmetic_v` on line 165.